templates/jobs/suspend-environments.groovy.j2

```diff
@@ -0,0 +1,88 @@
+job {
+  name '{{job_name}}'
+  description('Suspends environments')
+  parameters {
+    stringParam("EXCLUDE_ENV", "{{ exclude_env }}",
+            "Space separated list of environments to exclude from shutdown")
+  }
+  scm {
+    git {
+      remote {
+	    url('https://github.com/alphagov/tsuru-ansible.git')
+      }
+      branch('master')
+      createTag(false)
+    }
+  }
+  triggers {
+   cron("{{ cron_suspend_time }}")
+  }
+  wrappers {
+    colorizeOutput()
+  }
+  publishers {
+    mailer("{{ email_notification_address | default("the-multi-cloud-paas-team@digital.cabinet-office.gov.uk")}}")
+  }
+  steps {
+    shell('''#!/bin/bash
+# Disable output buffering to give realtime data
+export PYTHONUNBUFFERED=1
+
+# Assign SSH private key
+eval $(ssh-agent) && ssh-add ~/.ssh/insecure-deployer
+
+# Set up trap to clean up ssh-agent process in the event of any failure
+set -e
```

Comment:
I would say so. I have even considered doing shutdown of all environments in parallel (run the core of the for loop as a background job), but the output would be difficult to read. 

Solutions to run GCE even if AWS fails:
* remove set -e (perhaps set it up again after the loop)
* have a separate job for gce

Which one do you prefer?